plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -1480,3 +1480,49 @@ func (c *VolumeNodeChecker) predicate(pod *v1.Pod, meta algorithm.PredicateMetad
 	}
 	return true, nil, nil
 }
+
+type NoDeletedPVCChecker struct {
+	pvcInfo PersistentVolumeClaimInfo
+}
+
+// NewNoDeletedPVCsPredicate evaluates if a pod can fit due to the PVC it uses.
+func NewNoDeletedPVCsPredicate(pvcInfo PersistentVolumeClaimInfo) algorithm.FitPredicate {
+	c := &NoDeletedPVCChecker{
+		pvcInfo: pvcInfo,
+	}
+	return c.predicate
+}
+
+func (c *NoDeletedPVCChecker) predicate(pod *v1.Pod, meta algorithm.PredicateMetadata, nodeInfo *schedulercache.NodeInfo) (bool, []algorithm.PredicateFailureReason, error) {
+	if !utilfeature.DefaultFeatureGate.Enabled(features.PVCProtection) {
```

Comment:
Why is this feature gated? Isn't it a bug to schedule a pod whose PVC is being deleted?

Reply:
I would have thought so... I was surprised it was allowed today

Reply:
It changes behavior of scheduler. Currently it can schedule pods that use PVCs that are not bound, their PV counterpart does not exist, they're being deleted or the PVC itself doesn't exist at all.

Reply:
I removed the gate while moving the check from predicate to a non-node specific check.